reconcile/utils/gql.py

```diff
@@ -140,7 +149,11 @@ def query(self, query, variables=None, skip_validation=False):
         if "data" not in result:
             raise GqlApiError(("`data` field missing from GraphQL" "server response."))
 
-        return result["data"]
+        result_data = result["data"]
+        if self.test_data:
+            self.add_test_data(result_data)
+
+        return result_data
```

Comment:
so the query is really executed, to make sure the query syntax is correct but to make template tests reproducible, the actual data is replaced by the test data. that is already a lot better than what we do in qontract-reconcile with unit tests running on fixture only

Reply:
no longer relevant with the new approach